cli/azd/pkg/ioc/service_locator.go

```diff
@@ -0,0 +1,24 @@
+package ioc
+
+type ServiceLocator interface {
+	Resolve(instance any) error
+	ResolveNamed(name string, instance any) error
+}
+
+type serviceLocator struct {
+	container *NestedContainer
+}
+
+func NewServiceLocator(container *NestedContainer) ServiceLocator {
+	return &serviceLocator{
+		container: container,
+	}
+}
+
+func (s *serviceLocator) Resolve(instance any) error {
```

Comment:
I wish we could use `any` for really almost nothing. It's one of the things I don't feel comfortable using in Go.

If there is any way here to limit the types with some interface or something, I would vote for it.

Reply:
I would like more type safety here as well.  This is basically a limitation on Go generics support and an instance func is not able to use generic arguments.  The type of the `instance` is used to look up the type in the IoC container to be resolved.

Reply:
> This is basically a limitation on Go generics

Why do we want to live with this limitation?